### PR TITLE
Add education type column to certificates table 

### PR DESCRIPTION
**Pull request by djino97**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/321

Tests for mapper not added for now.

### CODE REVIEW

File: src/UserService.Mappers/Models/PatchDbUserMapper.cs [hunk 1]
```diff
@@ -30,13 +34,42 @@ public class PatchDbUserMapper : IPatchDbUserMapper
...
+                if (Regex.IsMatch(item.path, "/Certificates/.*?/Image"))
+                {
+                    var image = (ImageInfo)item.value;
+                    var numberElement = Regex.Match(item.path.Trim('/'), "/.*?/").Value;
+
+                    item.path = $"/{nameof(DbUser.Certificates)}{numberElement}{nameof(DbUserCertificate.ImageId)}";
+                    item.value = (Guid)getAvatarImageId.Invoke(image.Content);
+                }
+
+
```
Comment (KostinP) [inline src/UserService.Mappers/Models/PatchDbUserMapper.cs:72]: Extra line

File: src/UserService.Models.Dto/Models/Certificates/EditCertificate.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    public class EditCertificate
+    {
+        public Guid Id { get; set; }
+        public string Name { get; set; }
+        public string SchoolName { get; set; }
+        public EducationType EducationType { get; set; }
+        public DateTime ReceivedAt { get; set; }
+        public ImageInfo Image { get; set; }
+    }
+}
```
Comment (KostinP) [inline src/UserService.Models.Dto/Models/Certificates/EditCertificate.cs:15]: Model?

Comment (djino97) [reply src/UserService.Models.Dto/Models/Certificates/EditCertificate.cs:15]: yeaa

Comment (djino97) [reply src/UserService.Models.Dto/Models/Certificates/EditCertificate.cs:15]: Do you have suggestions?

File: src/UserService.Models.Dto/Models/CertificateInfo.cs [hunk 3]
```diff
@@ -1,12 +1,14 @@
-﻿using System;
+﻿using LT.DigitalOffice.UserService.Models.Dto.Enums;
+using System;
 
 namespace LT.DigitalOffice.UserService.Models.Dto.Models
 {
     public class CertificateInfo
     {
-        public Guid Id { get; set; }
+        public Guid? Id { get; set; }
```
[the remark this answers is not in the document]
Comment (djino97) [reply src/UserService.Models.Dto/Models/CertificateInfo.cs]: fixed

File: src/UserService.Models.Dto/Models/Certificates/EditCertificate.cs (hunk 2 above)
Comment (KostinP) [reply src/UserService.Models.Dto/Models/Certificates/EditCertificate.cs:15]: Request

File: src/UserService.Mappers/Models/PatchDbUserMapper.cs (hunk 1 above)
Comment (djino97) [reply src/UserService.Mappers/Models/PatchDbUserMapper.cs]: Fix

**Review by TIove — APPROVED** [review]

**Review by BelehovEgor — APPROVED** [review]

File: test/UserService.Mappers.UnitTests/PatchDbUserMapperTests.cs [hunk 4]
```diff
@@ -47,7 +78,33 @@ public void OneTimeSetUp()
...
+                new Operation<EditUserRequest>(
+                    "replace",
+                    $"/{nameof(DbUser.Certificates)}/0/{nameof(DbUserCertificate.Id)}",
+                    "",
+                    certificateId),
+                new Operation<EditUserRequest>(
+                    "replace",
+                    $"/{nameof(DbUser.Certificates)}/0/{nameof(DbUserCertificate.EducationType)}",
+                    "",
+                    1),
```
Comment (borovHokage) [inline test/UserService.Mappers.UnitTests/PatchDbUserMapperTests.cs:96]: why is 1 here, at the end?

**Review by borovHokage — APPROVED** [review]

**Review by KostinP — APPROVED** [review]

File: test/UserService.Data.UnitTests/UserRepositoryTests.cs [hunk 5]
```diff
@@ -26,12 +38,130 @@ public void OneTimeSetUp()
...
+                Name = "1С",
+                ReceivedAt = DateTime.UtcNow
+            };
+
+            _dbUser = new DbUser
+            {
+                Id = userId,
+                FirstName = "Ivan",
+                LastName = "Ivanovich",
+                MiddleName = "Ivanod",
```
[the remark this answers is not in the document]
Comment (djino97) [reply test/UserService.Data.UnitTests/UserRepositoryTests.cs]: Fixed

**Pull request merged**